File: backend/src/main/java/me/iamhardyha/bugbuddy/controller/dto/UserProfileResponse.java
package me.iamhardyha.bugbuddy.controller.dto;

import lombok.Getter;
import me.iamhardyha.bugbuddy.model.entity.UserEntity;

@Getter
public class UserProfileResponse {

    private final Long id;
    private final String nickname;
    private final String email;
    private final String bio;
    private final String role;
    private final String mentorStatus;
    private final int xp;
    private final int level;

    private UserProfileResponse(UserEntity user) {
        this.id = user.getId();
        this.nickname = user.getNickname();
        this.email = user.getEmail();
        this.bio = user.getBio();
        this.role = user.getRole().name();
        this.mentorStatus = user.getMentorStatus().name();
        this.xp = user.getXp();
        this.level = user.getLevel();
    }

    public static UserProfileResponse from(UserEntity user) {
        return new UserProfileResponse(user);
    }
}

File: backend/src/main/kotlin/me/iamhardyha/bugbuddy/controller/dto/UserProfileResponse.kt
package me.iamhardyha.bugbuddy.controller.dto

import me.iamhardyha.bugbuddy.model.entity.UserEntity

data class UserProfileResponse(
    val id: Long,
    val nickname: String,
    val email: String?,
    val bio: String?,
    val role: String,
    val mentorStatus: String,
    val xp: Int,
    val level: Int,
) {
    companion object {
        fun from(user: UserEntity) = UserProfileResponse(
            id = user.id!!,
            nickname = user.nickname,
            email = user.email,
            bio = user.bio,
            role = user.role.name,
            mentorStatus = user.mentorStatus.name,
            xp = user.xp,
            level = user.level,
        )
    }
}
